File: src/ros2_bridge/mappers/nav2_msgs/edge_cost.rs
//! Typed mapper for `nav2_msgs/msg/EdgeCost`.

use crate::ros2_bridge::mapper::TypedTopicMapper;

pub(crate) fn edge_cost_to_bus(msg: ros_env::nav2_msgs::msg::EdgeCost) -> crate::nav2_msgs::msg::v1::EdgeCost {
    crate::nav2_msgs::msg::v1::EdgeCost {
        edgeid: msg.edgeid,
        cost: msg.cost,
    }
}

pub(crate) fn edge_cost_to_ros(bus: crate::nav2_msgs::msg::v1::EdgeCost) -> ros_env::nav2_msgs::msg::EdgeCost {
    ros_env::nav2_msgs::msg::EdgeCost {
        edgeid: bus.edgeid,
        cost: bus.cost,
    }
}

#[derive(Clone, Copy, Debug, Default)]
pub struct Nav2MsgsEdgeCostMapper;

impl TypedTopicMapper for Nav2MsgsEdgeCostMapper {
    type Ros = ros_env::nav2_msgs::msg::EdgeCost;
    type Bus = crate::nav2_msgs::msg::v1::EdgeCost;

    fn type_name(&self) -> &'static str {
        "nav2_msgs/msg/EdgeCost"
    }

    fn ros_to_bus(&self, msg: Self::Ros) -> crate::errors::Result<Self::Bus> {
        Ok(edge_cost_to_bus(msg))
    }

    fn bus_to_ros(&self, msg: Self::Bus) -> crate::errors::Result<Self::Ros> {
        Ok(edge_cost_to_ros(msg))
    }
}

File: bindings/python/robot_bus/ros2_bridge/mappers/nav2_msgs/edge_cost.py
"""Generated mapper for `nav2_msgs/msg/EdgeCost`."""

from __future__ import annotations

from robot_bus.ros2_bridge.mappers import _convert

_BusMsg = None


def _bus_cls():
    global _BusMsg
    if _BusMsg is None:
        from robot_bus.nav2_msgs.msg.v1 import EdgeCost as BusMsg

        _BusMsg = BusMsg
    return _BusMsg


def edge_cost_to_bus(msg):
    BusMsg = _bus_cls()
    bus = BusMsg()
    bus.edgeid = msg.edgeid
    bus.cost = msg.cost
    return bus


def edge_cost_to_ros(bus):
    from nav2_msgs.msg import EdgeCost as RosMsg

    out = RosMsg()
    out.edgeid = bus.edgeid
    out.cost = bus.cost
    return out


class Nav2MsgsEdgeCostMapper:
    _ros_type = None

    def ros_msg_type(self):
        cls = type(self)
        if cls._ros_type is None:
            from nav2_msgs.msg import EdgeCost as RosMsg

            cls._ros_type = RosMsg
        return cls._ros_type

    def ros_to_bus(self, msg) -> bytes:
        return edge_cost_to_bus(msg).SerializeToString()

    def bus_to_ros(self, payload: bytes):
        BusMsg = _bus_cls()
        bus = BusMsg()
        bus.ParseFromString(payload)
        return edge_cost_to_ros(bus)
